compiler/ccgexprs.nim

```diff
@@ -1839,6 +1839,9 @@ proc genSomeCast(p: BProc, e: PNode, d: var TLoc) =
     if srcTyp.kind in {tyPtr, tyPointer} and etyp.kind in IntegralTypes:
       putIntoDest(p, d, e, "(($1) (ptrdiff_t) ($2))" %
           [getTypeDesc(p.module, e.typ), rdCharLoc(a)], a.storage)
+    elif p.config.selectedGc == gcDestructors and etyp.kind == tySequence:
+      putIntoDest(p, d, e, "(($1_Content*) ($2.p))" %
```

Comment:
Sorry, thinking about this more, this is still incorrect? The cast's result should still be (length, pointer) pair and not just the casted pointer.

Reply:
Yeah, (as I pointed out above). I'm working on a correct fix.